R/create.barplot.R

```diff
@@ -640,15 +640,17 @@ create.barplot <- function(
 					}
 
 				if (length(sample.order) == 1) {
-					if (sample.order == 'decreasing') {
-						# order is the new order the bars will appear in
-						ordering <- order(trellis.object$panel.args[[1]]$y[c(1:num.bars)]);
+					if(! sample.order %in% c('decreasing', 'increasing')) {
+						stop('sample.order should be `decreasing` or `increasing`');
 						}
 
-					# reverse order if increasing
-					if (sample.order == 'increasing') {
-						ordering <- rev(order(trellis.object$panel.args[[1]]$y[c(1:num.bars)]));
-						}
+					# This looks backwards but gets reversed later
+					# Might want to revisit if it makes more sense to sort in correct order here
+					sample.order.decreasing <- sample.order != 'decreasing';
```

Comment:
Should this be `sample.order == 'decreasing'` ?  

I think when `length(sample.order) == 1` and equals `increasing` or `decreasing`, then this line is always run twice: in line 649 and 710.  It seems the code is duplicated?

Reply:
The duplication is the same as it was originally in the code. One of them is in the block `plot.horizontal = FALSE` and the other is `plot.horizontal = TRUE`.

It "should" be `sample.order == 'decreasing'` but because of the way the code is implemented for `create.barplot` everything is reversed in order. I tried not to change too many sections and this was the most straightforward fix that kept all the other behavior the same but it would be worth refactoring some of this.